README.txt
LumoNet Google Maps

Extension Key: lumogooglemaps

Copyright 2006-2008 Thomas Off <typo3@retiolum.de>

This extension adds a map from Google Maps to your website. It is based on PHP GoogleMapAPI (http://www.phpinsider.com/php/code/GoogleMapAPI/) by Monte Ohrt which was adapted and partly rewritten to make it more suitable for usage in this extension (e.g. quoting of JavaScript code snippets).

Features:
* Show locations created as tt_address items
* Control the size of the map as well as other behaviour (e.g. whether map controls are shown or not)
* Templatable sidebar which lists the shown locations and can show them in detail once you click on them
* Templatable info box with support for links and graphics

See the manual in folder 'docs' or the online version at http://typo3.org for
more details about the extension and how to use it.

ChangeLog
(add new changes on top of this file)

2006-07-23 Thomas Off, LumoNet oHG <t.off@lumonet.de>

    * Temporarily fixed issue with getimagesize() in GoogleMapAPI.class.php

2006-07-20 Thomas Off, LumoNet oHG <t.off@lumonet.de>

    * Added option to set default zoom level for maps
    * Added option to set center point of maps
    * Improved geocoding using new Google Maps service
    * Added TS option to set default country for geocoding
    * Fixed dependencies from tt_address

2006-05-02 Thomas Off, LumoNet oHG <t.off@lumonet.de>

    * Initial release

ext_tables.php
<?php

if (!defined ('TYPO3_MODE')) {
    die ('Access denied.');
}

$tempColumns = Array (
    "tx_lumogooglemaps_longitude" => Array (
        "exclude" => 1,
        "label" => "LLL:EXT:lumogooglemaps/locallang_db.xml:tt_address.tx_lumogooglemaps_longitude",
        "config" => Array (
            "type" => "input",
            "size" => "20",
            "max" => "20",
            "eval" => "nospace",
        )
    ),
    "tx_lumogooglemaps_latitude" => Array (
        "exclude" => 1,
        "label" => "LLL:EXT:lumogooglemaps/locallang_db.xml:tt_address.tx_lumogooglemaps_latitude",
        "config" => Array (
            "type" => "input",
            "size" => "20",
            "max" => "20",
            "eval" => "nospace",
        )
    ),
);

t3lib_div::loadTCA("tt_address");
t3lib_extMgm::addTCAcolumns("tt_address", $tempColumns,1);
t3lib_extMgm::addToAllTCAtypes("tt_address", "tx_lumogooglemaps_longitude;;;;1-1-1, tx_lumogooglemaps_latitude");

t3lib_div::loadTCA('tt_content');
$TCA['tt_content']['types']['list']['subtypes_excludelist'][$_EXTKEY . '_pi1'] = 'layout,select_key,pages';

t3lib_extMgm::addPlugin(Array('LLL:EXT:' . $_EXTKEY . '/locallang_db.xml:tt_content.list_type_pi1', $_EXTKEY . '_pi1'), 'list_type');

t3lib_extMgm::addStaticFile($_EXTKEY, "pi1/static/", "Google map");

// Use FlexForms.
$TCA['tt_content']['types']['list']['subtypes_addlist'][$_EXTKEY . '_pi1'] = 'pi_flexform';
t3lib_extMgm::addPiFlexFormValue($_EXTKEY . '_pi1', 'FILE:EXT:' . $_EXTKEY . '/pi1/flexform_ds_pi1.xml');

if (TYPO3_MODE=="BE") {
    $TBE_MODULES_EXT["xMOD_db_new_content_el"]["addElClasses"]["tx_lumogooglemaps_pi1_wizicon"] = t3lib_extMgm::extPath($_EXTKEY) . 'pi1/class.tx_lumogooglemaps_pi1_wizicon.php';
}

?>

class.tx_lumogoolgemaps_mapapi.php
<?php
/*
 * Copyright notice
 *
 * (c) 2006-2008 Thomas Off <typo3@retiolum.de>
 * All rights reserved
 *
 * This script is part of the TYPO3 project. The TYPO3 project is
 * free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * The GNU General Public License can be found at
 * http://www.gnu.org/copyleft/gpl.html.
 *
 * This script is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * This copyright notice MUST APPEAR in all copies of the script!
 */

require_once(t3lib_extMgm::extPath('lumogooglemaps') . 'GoogleMapAPI.class.php');

/**
 * Class that extends and adapts the original GoogleMapAPI class for use within this extension.
 *
 * @author Thomas Off <typo3@retiolum.de>
 * @package TYPO3
 * @subpackage tx_lumogooglemaps
 */
class tx_lumogooglemaps_mapapi extends GoogleMapAPI {
}

?>

pi1/class.tx_lumogooglemaps_pi1_wizicon.php
<?php
/*
 * Copyright notice
 *
 * (c) 2006-2008 Thomas Off <typo3@retiolum.de>
 * All rights reserved
 *
 * This script is part of the TYPO3 project. The TYPO3 project is
 * free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * The GNU General Public License can be found at
 * http://www.gnu.org/copyleft/gpl.html.
 *
 * This script is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * This copyright notice MUST APPEAR in all copies of the script!
 */

/**
 * Class that adds the wizard icon.
 *
 * @author Thomas Off <typo3@retiolum.de>
 * @package TYPO3
 * @subpackage tx_lumogooglemaps
 */
class tx_lumogooglemaps_pi1_wizicon {

	/**
     * Processing the wizard items array
     *
     * @param array  $wizardItems: The wizard items
     * @return Modified array with wizard items
     */
    function proc($wizardItems)    {
        global $LANG;

        $LL = $this->includeLocalLang();

        $wizardItems['plugins_tx_lumogooglemaps_pi1'] = array(
            'icon' => t3lib_extMgm::extRelPath('lumogooglemaps') . 'pi1/ce_wiz.gif',
            'title' => $LANG->getLLL('pi1_title', $LL),
            'description' => $LANG->getLLL('pi1_plus_wiz_description', $LL),
            'params' => '&defVals[tt_content][CType]=list&defVals[tt_content][list_type]=lumogooglemaps_pi1',
        );

        return $wizardItems;
    }

    /**
     * Reads the [extDir]/locallang.xml and returns the \$LOCAL_LANG array found in that file.
     *
     * @return The array with language labels
     */
    function includeLocalLang()    {
        $llFile = t3lib_extMgm::extPath('lumogooglemaps') . 'locallang.xml';
        $LOCAL_LANG = t3lib_div::readLLXMLfile($llFile, $GLOBALS['LANG']->lang);
        return $LOCAL_LANG;
    }

}

if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/lumogooglemaps/pi1/class.tx_lumogooglemaps_pi1_wizicon.php'])    {
    include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/lumogooglemaps/pi1/class.tx_lumogooglemaps_pi1_wizicon.php']);
}

?>

ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext: "lumogooglemaps"
#
# Auto generated 23-07-2006 19:14
#
# Manual updates:
# Only the data in the array - anything else is removed by next write.
# "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
    'title' => 'LumoNet Google Maps',
    'description' => 'Add an interactive map (based on Google Maps) to your website.',
    'category' => 'plugin',
    'shy' => 0,
    'version' => '0.2.1',
    'dependencies' => 'tt_address',
    'conflicts' => '',
    'priority' => '',
    'loadOrder' => '',
    'module' => '',
    'state' => 'beta',
    'uploadfolder' => 1,
    'createDirs' => '',
    'modify_tables' => 'tt_address',
    'clearcacheonload' => 1,
    'lockType' => '',
    'author' => 'Thomas Off',
    'author_email' => 'typo3@retiolum.de',
    'author_company' => '',
    'CGLcompliance' => '',
    'CGLcompliance_note' => '',
    'constraints' => array(
        'depends' => array(
            'typo3' => '4.0',
            'php' => '',
            'tt_address' => '1.0.4',
        ),
        'conflicts' => array(
        ),
        'suggests' => array(
        ),
    ),
    '_md5_values_when_last_written' => 'a:29:{s:9:"ChangeLog";s:4:"111d";s:22:"GoogleMapAPI.class.php";s:4:"2b27";s:10:"README.txt";s:4:"bc2f";s:12:"ext_icon.gif";s:4:"236a";s:17:"ext_localconf.php";s:4:"caa8";s:14:"ext_tables.php";s:4:"36c9";s:14:"ext_tables.sql";s:4:"3d4e";s:28:"ext_typoscript_constants.txt";s:4:"3f20";s:24:"ext_typoscript_setup.txt";s:4:"fcdc";s:13:"locallang.xml";s:4:"b6bc";s:16:"locallang_db.xml";s:4:"472c";s:14:"doc/manual.sxw";s:4:"4faa";s:14:"pi1/ce_wiz.gif";s:4:"5491";s:35:"pi1/class.tx_lumogooglemaps_pi1.php";s:4:"57a6";s:43:"pi1/class.tx_lumogooglemaps_pi1_wizicon.php";s:4:"9eac";s:13:"pi1/clear.gif";s:4:"cc11";s:23:"pi1/flexform_ds_pi1.xml";s:4:"75e5";s:17:"pi1/locallang.xml";s:4:"2a3d";s:25:"templates/marker_icon.png";s:4:"1f92";s:27:"templates/marker_shadow.png";s:4:"f77b";s:27:"templates/template_css.html";s:4:"2018";s:25:"templates/.svn/README.txt";s:4:"cf1a";s:25:"templates/.svn/empty-file";s:4:"d41d";s:22:"templates/.svn/entries";s:4:"7eeb";s:21:"templates/.svn/format";s:4:"48a2";s:44:"templates/.svn/prop-base/marker.gif.svn-base";s:4:"1131";s:40:"templates/.svn/props/marker.gif.svn-work";s:4:"1131";s:44:"templates/.svn/text-base/marker.gif.svn-base";s:4:"1990";s:51:"templates/.svn/text-base/template_css.html.svn-base";s:4:"e058";}',
    'suggests' => array(
    ),
);

?>